transport/src/main/java/io/netty/channel/DefaultFileRegion.java

```diff
@@ -100,65 +135,60 @@ public boolean isOpen() {
     public void open() throws IOException {
         if (!isOpen() && refCnt() > 0) {
             // Only open if this DefaultFileRegion was not released yet.
-            file = new RandomAccessFile(f, "r").getChannel();
+            fileChannel = new RandomAccessFile(file, "r").getChannel();
         }
     }
 
     @Override
-    public long position() {
-        return position;
+    public FileChannel channel() throws IOException {
+        // Call open to make sure fc is initialized. This is a no-op if we called it before.
+        open();
+        return fileChannel;
     }
 
     @Override
-    public long count() {
-        return count;
+    public FileRegion skipBytes(long length) {
+        super.skipBytes(length);
+        return this;
     }
 
     @Override
-    public long transfered() {
-        return transfered;
+    public ReadableObject slice(long position, long length) {
+        ensureReadable(length);
```

Comment:
Is this necessary? Shouldn't we just cover this in the next if statement?  We should be considering the `position` parameter when checking validity too.